packages/core-data/src/actions.js

```diff
@@ -155,11 +161,21 @@ export function receiveEmbedPreview( url, preview ) {
 export function* deleteEntityRecord( kind, name, recordId, query ) {
 	const entities = yield getKindEntities( kind );
 	const entity = find( entities, { kind, name } );
-	let error;
-	let deletedRecord = false;
 	if ( ! entity ) {
 		return;
 	}
+	const lock = yield acquireStoreLock( [ kind, name, recordId ], true );
```

Comment:
Minor, but boolean arguments really irk me as you have to remember or look up what the argument is. Could we do `acquireStoreLock( [ kind, name, recordId ], { exclusive: true } )`?